<!--
-->
<launch>

  # Bag filename
  <arg name="bag_file" />
  <arg name="start" default="0"/>
  <arg name="camera" default="/front"/>
  <arg name="robot_frame" default="/base_link"/>
  <arg name="tolerance" default="0.15"/>
  
  # Use simulation time 
  <param name="use_sim_time" value="true" />

  # Bag file
<!--   <node name="bag_player" type="play" pkg="rosbag" args="$(arg bag_file) - -clock -s $(arg start)" /> -->
  
  # Uncompress depth
  <node name="republish_depth" type="republish" pkg="image_transport" args="compressedDepth in:=$(arg camera)/depth_registered/image_raw raw out:=$(arg camera)/depth_registered/image_raw"/>
  
  # Launch plane detector
  <node name="test_floor_detector" type="test_floor_detector" pkg="plane_detector" output="screen" >
   <param name="camera" value="$(arg camera)"/>
   <param name="link_1" value="$(arg robot_frame)"/>
   <param name="link_2" value="$(arg camera)_depth_optical_frame"/>
   <param name="floor_tolerance" value="$(arg tolerance)"/>
  </node>
  
  # TFs of the camera
  <roslaunch file="$(find openni2_launch)/launch/openni2.launch">
    <arg name="load_driver" value="false"/>
    <arg camera="$(arg camera)/depth"/>
  </roslaunch>
  
<!--   <node name="rviz" type="rviz" pkg="rviz"/> -->
  
</launch>
